TempForceSensor/README.md

```diff
@@ -0,0 +1,234 @@
+- UP Board
+- 1 breadboard and a couple of wires
+- HDMI Monitor
+
+## Parts Review
+
+* MCP3002 or MCP3208
+
+Below are the pinouts of the MCP3002 and MCP3208 analog-to-digital converters (ADC) used in this sample.
+
+![Electrical Components](../Resources/MCP3002.png)
+![Electrical Components](../Resources/MCP3208.png)
+
+* UP Board
+
+ ![UP Board Pinout](../Resources/Upboard_Pinout.png)
+
+## Parts Connection
+
+1. Connect the TMP36 temperature Sensor to the MCP3002; `Sensor output pin` (the middle pin) should be connected to `CH0` on the MCP3002;
+
+If you are using a [Force sensor](http://www.digikey.com/product-detail/en/30-81794/1027-1001-ND/2476468) which only has two legs, set the left leg to 5V,
+and connect the other Leg to `CH0` on MCP3002
+
+Detailed connection:
+
+![Overall Schematics](../Resources/temp_mcp3002.png);
+![Overall Schematics](../Resources/force_mcp3002.png);
+
+With each model of UP Board, the pin layout might be a little different. But the pin connection with MCP3002 should be as below:
+
+- MCP3002: VDD/VREF - 5V on UP Board
+- MCP3002: CLK - "SPI2 SCLK" on UP Board
+- MCP3002: Dout - "SPI2 MISO" on UP Board
+- MCP3002: Din - "SPI2 MOSI" on UP Board
+- MCP3002: CS/SHDN - "SPI2 CS0" on UP Board
+- MCP3002: DGND - GND on UP Board
+- MCP3002: CH0- Sensor Output Pin
+
+2. **Alternative: If you are using MCP3208** Connect the temperature Sensor to MCP3208; `Sensor output pin` (the mid pin) should be connected to `CH0` on MCP3208.
+
+Detailed connection:
+
+![Overall Schematics](../Resources/OverallCon_mcp3208.png )
+
+With each model of UP Board, the pin layout might be a little different.
+But the pin connection with MCP3208 should be as below:
+
+- MCP3208: VDD - 5V on UP Board
+- MCP3208: VREF - 5V on UP Board
+- MCP3208: CLK - "SPI2 SCLK" on UP Board
+- MCP3208: Dout - "SPI2 MISO" on UP Board
+- MCP3208: Din - "SPI2 MOSI" on UP Board
+- MCP3208: CS/SHDN - "SPI2 CS0 on UP Board
+- MCP3208: DGND - GND on UP Board
+
+## Look at the code
+
+Let's go through the code. We use a timer in the sample, and each time the 'Tick' event is called,
+we read the sensor data through ADC, and the value will be displayed on the screen.
+
+* Timer Code
+Setup timer in C#:
+```csharp
+public MainPage()
+{
+	// ...
+
+	this.timer = new DispatcherTimer();
+	this.timer.Interval = TimeSpan.FromMilliseconds(500);
+	this.timer.Tick += Timer_Tick;
+	this.timer.Start();
+
+	// ...
+}
+private void Timer_Tick(object sender, object e)
+{
+	DisplayTextBoxContents();
+}
+```
+
+* Initialize SPI pin
+```csharp
+private async void InitSPI()
+{
+    try
+    {
+        var settings = new SpiConnectionSettings(SPI_CHIP_SELECT_LINE);
+        settings.ClockFrequency = 1000000;// For UP Board use 1MHzs and For Rpi use 500KHzs
+        settings.Mode = SpiMode.Mode0; //Mode3;
+
+        string spiAqs = SpiDevice.GetDeviceSelector(SPI_CONTROLLER_NAME);

```

Comment:
The actual code doesn't use this pattern.  Instead it uses SpiDevice.GetDefaultAsync() and assumes that the device returned is the correct one even though the source code has an "SPI_CONTOLLER_NAME" constant define.  Should this pattern be applied to the mainpage.xaml.cs file?

Reply:
Available SPI devices gets recognize automatically in UPBOARD
SpiDevice.GetDeviceSelector();

As UPBOARD has SPI2 mode available, so we are updating that in the code.